### PR TITLE
expose Path.contains_points as a method of Patch

### PR DESCRIPTION
**Pull request by yymao**
## PR Summary

`Patch` has the method `contains_point` but not `contains_points`, but the latter is supported by `Path`. This PR adds `contains_points` to `Patch`.

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is PEP 8 compliant
- [x] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/patches.py [hunk 1]
```diff
@@ -181,14 +181,29 @@ def contains(self, mouseevent, radius=None):
 
     def contains_point(self, point, radius=None):
         """
-        Returns *True* if the given point is inside the path
+        Returns `True` if the given point is inside the path
```
Comment (dstansby) [inline lib/matplotlib/patches.py]: Sorry to be a pain:
- Bits of code (like True) should have double ticks around them (``True``)
- Keyword arguments (like points below) should have stars (*points*)

**Comment by yymao** [comment]
@dstansby thanks for the review. I've updated the docstring to suggested format.

**Comment by dstansby** [comment]
It looks like there's a few more PEP8 errors:

```
/home/travis/build/matplotlib/matplotlib/lib/matplotlib/patches.py:186:1: W293 blank line contains whitespace

^

/home/travis/build/matplotlib/matplotlib/lib/matplotlib/patches.py:196:80: E501 line too long (99 > 79 characters)

        Returns a bool array which is `True` if the (closed) path contains the corresponding point.

                                                                               ^

/home/travis/build/matplotlib/matplotlib/lib/matplotlib/patches.py:198:1: W293 blank line contains whitespace

^

/home/travis/build/matplotlib/matplotlib/lib/matplotlib/patches.py:199:43: W291 trailing whitespace

        *points* should be an N x 2 array.
```

**Comment by yymao** [comment]
All fixed now. Thanks @dstansby.

File: lib/matplotlib/patches.py [hunk 2]
```diff
@@ -181,14 +181,30 @@ def contains(self, mouseevent, radius=None):
...
                                               self.get_transform(),
                                               radius)
 
+    def contains_points(self, points, radius=None):
+        """
+        Returns a bool array which is `True` if the (closed) path
+        contains the corresponding point.
+        (transformed with its transform attribute).
+
+        *points* should be an N x 2 array.
```
Comment (jklymak) [inline lib/matplotlib/patches.py]: Do you need to check for the right shape before sending to contains_points?  Can you pass a 2xN array?  Thats what Id usually pass...

Comment (yymao) [reply lib/matplotlib/patches.py]: @jklymak This PR only exposes the `Path.contains_points()` method, so I think further improvement to this method should be separately made to https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/path.py#L501

Comment (jklymak) [reply lib/matplotlib/patches.py]: Yes but the docs shouldn't say something vague like "usually"

Comment (yymao) [reply lib/matplotlib/patches.py]: @jklymak that's fair. I changed to "must be", to be consistent with what `Path.contains_points()` raises when 2xN array is passed.

**Comment by yymao** [comment]
@jklymak I am not sure what's the best thing to do here. The `Path.contains_points()` method is already tested in `lib/matplotlib/tests/test_path.py`. Here we just expose this method to `Patch` with `Patch.get_path()`. If we want to test this exposure, I guess we should add a test in `lib/matplotlib/tests/test_patches.py`? I did not see a similar test for `Patch..contains_point()` there so I am not sure (sorry, new contributor here). Suggestions welcome.

**Comment by jklymak** [comment]
@yymao I think either place would be fine.  

Its prob. not the most crucial test to have, but a) it would prove your PR works, and b) it would protect the functionality in the future.  (i.e. if someone breaks it somehow, the test will catch it).

**Comment by yymao** [comment]
@jklymak thanks, I've added `test_contains_point()` and `test_contains_points()` to `lib/matplotlib/tests/test_patches.py`. These tests are fairly simple, only to test if the these methods of `Path` are properly ported to `Patch`, as more involved tests are already in `lib/matplotlib/tests/test_path.py`.

**Comment by yymao** [comment]
Happy new year all. I'm pinging again to see if there's any other things I can help with this PR, otherwise I think this PR is ready to be merged. Thanks!

File: lib/matplotlib/patches.py [hunk 3]
```diff
@@ -181,14 +181,30 @@ def contains(self, mouseevent, radius=None):
 
     def contains_point(self, point, radius=None):
         """
-        Returns *True* if the given point is inside the path
+        Returns `True` if the given *point* is inside the path
         (transformed with its transform attribute).
```
Comment (jklymak) [inline lib/matplotlib/patches.py]: True should be double left quotes, "``" here and below....

Comment (yymao) [reply lib/matplotlib/patches.py]: Actually, by double left quotes do you mean \`\`True\`\` or \`True\`? 

It was \`True\` already, and now I've updated them to \`\`True\`\` but feel that I should confirm this...

Comment (jklymak) [reply lib/matplotlib/patches.py]: The updated version is what I meant.  But your tests are still broken, and it looks like you may have to rebase on master to resolve the conflict in `test_patches.py`.  If you need help w/ that, let us know; I think it only took my 10 tries until I figured out how to do it properly (basically rebase all your commits *first* and then rebase onto master...)

Comment (yymao) [reply lib/matplotlib/patches.py]: Yes, aware of the test failing but I couldn't figure out why...

Comment (jklymak) [reply lib/matplotlib/patches.py]: The error messages are in the logs of the tests.  Just need to click through.

If you need to debug on your local machine (reccomended), then have a look at https://matplotlib.org/devdocs/devel/testing.html  If anything is unclear in there let us know!

**Comment by jklymak** [comment]
I'll merge once all the tests have passed.  ping me if I forget.  Thanks for your work on this!

**Comment by yymao** [comment]
@jklymak thanks for guiding a new contributor here. Much appreciated!

**Comment by jklymak** [comment]
@yymao.  My pleasure!  Hope you contribute again!

**Pull request merged**